### ISSUE TITLE
tsibble::as_tibble.tbl_ts does not preserve groups

### ISSUE THREAD

**Issue by wkdavis** [tidyverts/tsibble#211]
The `tsibble::as_tibble()` method for `tbl_ts` does not preserve groupings from the input tsibble. I think it would be a useful feature for groups to be preserved.

```r
library(tsibble)
library(tsibbledata)
library(dplyr)

> aus_livestock %>% group_by_key() -> livestock_grp

> livestock_grp
# A tsibble: 29,364 x 4 [1M]
# Key:       Animal, State [54]
# Groups:    Animal, State [54]
      Month Animal                     State                        Count
      <mth> <fct>                      <fct>                        <dbl>
 1 1976 Jul Bulls, bullocks and steers Australian Capital Territory  2300
 2 1976 Aug Bulls, bullocks and steers Australian Capital Territory  2100
 3 1976 Sep Bulls, bullocks and steers Australian Capital Territory  2100
 4 1976 Oct Bulls, bullocks and steers Australian Capital Territory  1900
 5 1976 Nov Bulls, bullocks and steers Australian Capital Territory  2100
 6 1976 Dec Bulls, bullocks and steers Australian Capital Territory  1800
 7 1977 Jan Bulls, bullocks and steers Australian Capital Territory  1800
 8 1977 Feb Bulls, bullocks and steers Australian Capital Territory  1900
 9 1977 Mar Bulls, bullocks and steers Australian Capital Territory  2700
10 1977 Apr Bulls, bullocks and steers Australian Capital Territory  2300
# … with 29,354 more rows

> tsibble::as_tibble(livestock_grp)
# A tibble: 29,364 x 4
      Month Animal                     State                        Count
      <mth> <fct>                      <fct>                        <dbl>
 1 1976 Jul Bulls, bullocks and steers Australian Capital Territory  2300
 2 1976 Aug Bulls, bullocks and steers Australian Capital Territory  2100
 3 1976 Sep Bulls, bullocks and steers Australian Capital Territory  2100
 4 1976 Oct Bulls, bullocks and steers Australian Capital Territory  1900
 5 1976 Nov Bulls, bullocks and steers Australian Capital Territory  2100
 6 1976 Dec Bulls, bullocks and steers Australian Capital Territory  1800
 7 1977 Jan Bulls, bullocks and steers Australian Capital Territory  1800
 8 1977 Feb Bulls, bullocks and steers Australian Capital Territory  1900
 9 1977 Mar Bulls, bullocks and steers Australian Capital Territory  2700
10 1977 Apr Bulls, bullocks and steers Australian Capital Territory  2300
# … with 29,354 more rows

# Groupings must be re-applied manually
> tsibble::as_tibble(livestock_grp) %>% group_by(across(group_vars(livestock_grp)))
# A tibble: 29,364 x 4
# Groups:   Animal, State [54]
      Month Animal                     State                        Count
      <mth> <fct>                      <fct>                        <dbl>
 1 1976 Jul Bulls, bullocks and steers Australian Capital Territory  2300
 2 1976 Aug Bulls, bullocks and steers Australian Capital Territory  2100
 3 1976 Sep Bulls, bullocks and steers Australian Capital Territory  2100
 4 1976 Oct Bulls, bullocks and steers Australian Capital Territory  1900
 5 1976 Nov Bulls, bullocks and steers Australian Capital Territory  2100
 6 1976 Dec Bulls, bullocks and steers Australian Capital Territory  1800
 7 1977 Jan Bulls, bullocks and steers Australian Capital Territory  1800
 8 1977 Feb Bulls, bullocks and steers Australian Capital Territory  1900
 9 1977 Mar Bulls, bullocks and steers Australian Capital Territory  2700
10 1977 Apr Bulls, bullocks and steers Australian Capital Territory  2300
# … with 29,354 more rows

```

I think this is a method dispatch issue, because if I call `tsibble:::as_tibble.grouped_ts(livestock_grp)` directly it works as expected.

```r
> tsibble:::as_tibble.grouped_ts(livestock_grp)
# A tibble: 29,364 x 4
# Groups:   Animal, State [54]
      Month Animal                     State                        Count
      <mth> <fct>                      <fct>                        <dbl>
 1 1976 Jul Bulls, bullocks and steers Australian Capital Territory  2300
 2 1976 Aug Bulls, bullocks and steers Australian Capital Territory  2100
 3 1976 Sep Bulls, bullocks and steers Australian Capital Territory  2100
 4 1976 Oct Bulls, bullocks and steers Australian Capital Territory  1900
 5 1976 Nov Bulls, bullocks and steers Australian Capital Territory  2100
 6 1976 Dec Bulls, bullocks and steers Australian Capital Territory  1800
 7 1977 Jan Bulls, bullocks and steers Australian Capital Territory  1800
 8 1977 Feb Bulls, bullocks and steers Australian Capital Territory  1900
 9 1977 Mar Bulls, bullocks and steers Australian Capital Territory  2700
10 1977 Apr Bulls, bullocks and steers Australian Capital Territory  2300
# … with 29,354 more rows
```

This is definitely not my area of expertise, but I think the issue might be that calling `tsibble::as_tibble()` is simply a reexport of `tibble::as_tibble()`, which will then start looking for a method for `as_tibble()` against the classes of the object. In this example:

```r
> class(livestock_grp)
[1] "grouped_ts" "grouped_df" "tbl_ts"     "tbl_df"     "tbl"        "data.frame"
```

However, because `tsibble:::as_tibble.grouped_ts()` is not exported from `tsibble`, `tibble`'s call to `UseMethod("tibble")` cannot find it in the namespace, and proceeds to the next class for which it can find a method, which would be `tbl_ts`, and in `tsibble::as_tibble.tbl_ts()` there is no preservation of groups. I would assume the way to resolve this, if desired, would just be to export `as_tibble.grouped_ts()`.

**Comment by earowang** [comment]
This behaviour is consistent with `as_tibble(<grouped_df>)` for dropping 
grouping information. You can find the reasoning behind here 
https://github.com/rnabioco/valr/issues/338#issuecomment-437584810, back 
to a bare tibble. I'd suggest to use `grouped_df()` to coerce to 
<grouped_df> if needed.

```r
library(dplyr)
#>
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#>
#>     filter, lag
#> The following objects are masked from 'package:base':
#>
#>     intersect, setdiff, setequal, union
mtcars %>%
   group_by(cyl) %>%
   as_tibble()
#> # A tibble: 32 x 11
#>      mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
#>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
#>  1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
#>  2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
#>  3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
#>  4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
#>  5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
#>  6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
#>  7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
#>  8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
#>  9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
#> 10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
#> # … with 22 more rows

library(tsibble)
ped_grp <- pedestrian %>%
   group_by_key()
ped_grp %>%
   grouped_df(vars = group_vars(ped_grp))
#> # A tibble: 66,037 x 5
#> # Groups:   Sensor [4]
#>    Sensor         Date_Time           Date        Time Count
#>    <chr>          <dttm>              <date>     <int> <int>
#>  1 Birrarung Marr 2015-01-01 00:00:00 2015-01-01     0  1630
#>  2 Birrarung Marr 2015-01-01 01:00:00 2015-01-01     1   826
#>  3 Birrarung Marr 2015-01-01 02:00:00 2015-01-01     2   567
#>  4 Birrarung Marr 2015-01-01 03:00:00 2015-01-01     3   264
#>  5 Birrarung Marr 2015-01-01 04:00:00 2015-01-01     4   139
#>  6 Birrarung Marr 2015-01-01 05:00:00 2015-01-01     5    77
#>  7 Birrarung Marr 2015-01-01 06:00:00 2015-01-01     6    44
#>  8 Birrarung Marr 2015-01-01 07:00:00 2015-01-01     7    56
#>  9 Birrarung Marr 2015-01-01 08:00:00 2015-01-01     8   113
#> 10 Birrarung Marr 2015-01-01 09:00:00 2015-01-01     9   166
#> # … with 66,027 more rows
```

<sup>Created on 2020-07-21 by the [reprex 
package](https://reprex.tidyverse.org) (v0.3.0)</sup>
Cheers,
Earo

**Comment by wkdavis** [comment]
Okay that makes sense, I will use `grouped_df`. I noticed that this seems to contradict [as_tibble.grouped_ts()](https://github.com/tidyverts/tsibble/blob/0ab33f996f65b860ce4c5435cc4b78191bdf8c07/R/as-tsibble.R#L499-L502):

```r
as_tibble.grouped_ts <- function(x, ...) {
  new_tibble(vec_data(x), nrow = nrow(x),
    groups = group_data(x), class = "grouped_df")
}
```

I found this function and noticed the `groups` argument, so I assumed it would maintain groups. Is that a bug in that function, or am I reading it wrong?
